### ISSUE TITLE
extraAttributes pull wrong value incorrectly

### ISSUE THREAD

**Issue by Ebag333** [pyfa-org/Pyfa#860]
Cross posted from #812 

From `saveddata.fit`:
```
    @property
    def tank(self):
        hps = {"passiveShield" : self.calculateShieldRecharge()}
        for type in ("shield", "armor", "hull"):
            hps["%sRepair" % type] = self.extraAttributes["%sRepair" % type]

        return hps
```

Turns out that this is using the _base_ amounts.  So for example, a small AAR on an unbonused ship with V skills and Improved Exile will give 78 HP for `armorDamageAmount` (current/actual).  If, however, you look at `self.extraAttributes`, you get 52 (base).

Digging into it, I can see why.

`self.extraAttributes` is set when we run effects.  For example
```
# shieldBoosting
#
# Used by:
# Modules from group: Shield Booster (93 of 93)
runTime = "late"
type = "active"
def handler(fit, module, context):
    amount = module.getModifiedItemAttr("shieldBonus")
    speed = module.getModifiedItemAttr("duration") / 1000.0
    fit.extraAttributes.increase("shieldRepair", amount / speed)
```

While this one is late, not all of them are.  Plus, if we set something else to run late (like maybe something that modifies boost amount) there's nothing that forces them to run before this effect runs.

TL;DR: can't trust the `extraAttributes`.  Awesome.  This is probably a bigger issue with _many_ things we set in `extraAttributes` and not just repairs.

**Comment by blitzmann** [comment]
The extra attributes are for pyfa-specific things that don't have an attribute attached to an item. I'm on Monday Ile right now, but I'm having trouble understanding what the problem is here? Is there something incorrect with the stats that we show the user?

**Comment by Ebag333** [comment]
Specifically, the issue I ran across is that AAR uses the base value, not the modified value.  (And to make matters worse, AARs use their base rep value as the HP/s value, not the `modified rep / duration value`.)

I _believe_ that it is happening because we set `extraAttributes` in an effect, and other effects that modify that value may run after the effect that sets it is run.

So yes, in some situations we are showing the wrong numbers.

There's other issues with the way we show HP/s that revolve around our base number for the reinforced (or theoretical maximum) tank using the base rep amount for the base HP/s (like not handling reloads).

TL;DR: AARs are super badly broken, and any `extraAttribute` that's set in an effect could be set before other things modify it.

**Comment by Ebag333** [comment]
Okay, found the particular issue for AARs.  I can submit a separate PR to fix that one....

**Comment by blitzmann** [comment]
Please give me a fit and use case in which values are off and provide expected vs actual values.

**Comment by Ebag333** [comment]
I suspect (but have not proven) that this is what causes the issue fixed in #770
